File: CollabTodoDesktop/Models/Notification.cs
using System;

namespace CollabTodoDesktop.Models;

/// <summary>
/// 알림 도메인 모델
/// Python 버전의 Notification dataclass와 동일한 구조
/// </summary>
public record Notification
{
    public int Id { get; init; }
    public int RecipientId { get; init; }
    public int? TaskId { get; init; }
    public string NotificationType { get; init; } = string.Empty;
    public string Message { get; init; } = string.Empty;
    public bool IsRead { get; init; }
    public DateTime CreatedAt { get; init; }
    public DateTime? ReadAt { get; init; }
}

File: collab_todo/models/notification.py
"""알림 도메인 모델."""

from __future__ import annotations

from dataclasses import dataclass
from datetime import datetime
from typing import Optional


@dataclass(frozen=True)
class Notification:
    id: int
    recipient_id: int
    task_id: Optional[int]
    notification_type: str
    message: str
    is_read: bool
    created_at: datetime
    read_at: Optional[datetime]
